packages/jsii/lib/assembler.ts

```diff
@@ -337,11 +337,15 @@ export class Assembler implements Emitter {
         return [jsiiType];
     }
 
-    private async _visitClass(type: ts.Type, namespace: string[]): Promise<spec.ClassType> {
+    private async _visitClass(type: ts.Type, namespace: string[]): Promise<spec.ClassType | undefined> {
```

Comment:
I don't think we're taking care of the `@internal` base-class case here. In such cases, you need to look at the base's base until there is none left.

Reply:
I believe we do. I am adding a test to that extent. The error will be "Unable to resolve referenced type 'jsii.InternalBaseClass'. Missing export?"